lib/tasks/generateResourcesJson.js

```diff
@@ -97,12 +97,20 @@ function getCreatorOptions(projectName) {
  * @param {object} parameters Parameters
  * @param {module:@ui5/fs.DuplexCollection} parameters.workspace DuplexCollection to read and write files
  * @param {module:@ui5/fs.AbstractReader} [parameters.dependencies] Reader or Collection to read dependency files
+ * @param {module:@ui5/builder.tasks.TaskUtil|object} [parameters.taskUtil] TaskUtil
  * @param {object} parameters.options Options
  * @param {string} parameters.options.projectName Project name
  * @returns {Promise<undefined>} Promise resolving with <code>undefined</code> once data has been written
  */
-module.exports = async function({workspace, dependencies, options: {projectName}}) {
-	const resources = await workspace.byGlob(["/resources/**/*"].concat(DEFAULT_EXCLUDES));
+module.exports = async function({workspace, dependencies, taskUtil, options: {projectName}}) {
+	let resources = await workspace.byGlob(["/resources/**/*"].concat(DEFAULT_EXCLUDES));
+
+	if (taskUtil) {
+		// Filter out resources that will be omitted from the build results
```

Comment:
What about `dependencies` a few lines below?

Maybe make the dependencies parameter mandatory as advised in the comment (breaking) and create a combo locator as done in other tasks as well.

We probably need to promote the OmitFromBuildResult tag to a global one then: https://github.com/SAP/ui5-builder/blob/next/lib/builder/BuildContext.js#L6

Reply:
Resources of dependencies are only used for dependency analysis so that required modules are listed correctly.
So at least for listing of the resources, this is not relevant.

But to be consistent, we should probably also filter the resources of the dependencies.

Reply:
I saw the 3.0 comment but just wanted to solve this issue first. But it seems to make sense to just do that right away.

Reply:
Btw, we still need to pass the resources separately, so AFAICS a combo locator can't be used.

Reply:
Done. I will add the changes to the [FIX] commit once the PR is reviewed.

Reply:
Sounds good 👍